### PR TITLE
[AMORO-2614]Get `PartitionSpec` based on the specId carried by the file

### LINKED ISSUE #2614: [Bug]: Failed to view snapshot details
### What happened?

image
After dropping a parition field, we cannot view the details of the snapshot generated by the delete operation.

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

1. create table ... partition by (bucket(id,16))
2. append data
3. drop partition field bucket(id,16)
4. enable data-expire and clean old data
5. view the details of the snapshot generated by the delete operation

### Relevant log output

```shell
2024-03-05 10:17:34,242 ERROR [JettyServerThreadPool-229412] [com.netease.arctic.server.dashboard.DashboardServer] [] - Failed to handle request                                                              
java.lang.IllegalArgumentException: Wrong class, expected java.lang.Long, but was java.lang.Integer, for object: 28                                                                                           
        at org.apache.iceberg.PartitionData.get(PartitionData.java:117) ~[iceberg-core-1.4.3.jar:?]                                                                                                           
        at org.apache.iceberg.PartitionSpec.get(PartitionSpec.java:169) ~[iceberg-api-1.4.3.jar:?]                                                                                                            
        at org.apache.iceberg.PartitionSpec.partitionToPath(PartitionSpec.java:187) ~[iceberg-api-1.4.3.jar:?]                                                                                                
        at com.netease.arctic.server.dashboard.MixedAndIcebergTableDescriptor.lambda$getSnapshotDetail$8(MixedAndIcebergTableDescriptor.java:343) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]                    
        at org.apache.iceberg.relocated.com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[iceberg-bundled-guava-1.4.3.jar:?]                                                          
        at com.netease.arctic.server.dashboard.MixedAndIcebergTableDescriptor.getSnapshotDetail(MixedAndIcebergTableDescriptor.java:336) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]                             
        at com.netease.arctic.server.dashboard.ServerTableDescriptor.getSnapshotDetail(ServerTableDescriptor.java:86) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]                                                
        at com.netease.arctic.server.dashboard.controller.TableController.getSnapshotDetail(TableController.java:368) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]                                                
        at io.javalin.core.security.SecurityUtil.noopAccessManager(SecurityUtil.kt:20) ~[javalin-4.6.1.jar:4.6.1]                                                                                             
        at io.javalin.http.JavalinServlet.addHandler$lambda-0(JavalinServlet.kt:96) ~[javalin-4.6.1.jar:4.6.1]                                                                                                
        at io.javalin.http.JavalinServlet$lifecycle$2$1$1.invoke(JavalinServlet.kt:43) ~[javalin-4.6.1.jar:4.6.1]                                                                                             
        at io.javalin.http.JavalinServlet$lifecycle$2$1$1.invoke(JavalinServlet.kt:43) ~[javalin-4.6.1.jar:4.6.1]                                                                                             
        at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:98) ~[javalin-4.6.1.jar:4.6.1]                                                                                      
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                       
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995) ~[?:1.8.0_291]                                                                                                  
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137) ~[?:1.8.0_291]                                                                                                     
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                        
        at io.javalin.http.JavalinServletHandler.executeNextTask$lambda-10$lambda-9(JavalinServletHandler.kt:112) ~[javalin-4.6.1.jar:4.6.1]                                                                  
        at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616) ~[?:1.8.0_291]                                                                                                         
        at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628) ~[?:1.8.0_291]                                                                                                    
        at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996) ~[?:1.8.0_291]                                                                                                       
        at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:112) ~[javalin-4.6.1.jar:4.6.1]                                                                                     
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                       
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995) ~[?:1.8.0_291]                                                                                                  
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137) ~[?:1.8.0_291]                                                                                                     
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                        
        at io.javalin.http.JavalinServletHandler.executeNextTask$lambda-10$lambda-9(JavalinServletHandler.kt:112) ~[javalin-4.6.1.jar:4.6.1]                                                                  
        at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616) ~[?:1.8.0_291]                                                                                                         
        at java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:628) ~[?:1.8.0_291]                                                                                                    
        at java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:1996) ~[?:1.8.0_291]                                                                                                       
        at io.javalin.http.JavalinServletHandler.executeNextTask(JavalinServletHandler.kt:112) ~[javalin-4.6.1.jar:4.6.1]                                                                                     
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$lambda-1(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                       
        at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995) ~[?:1.8.0_291]                                                                                                  
        at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137) ~[?:1.8.0_291]                                                                                                     
        at io.javalin.http.JavalinServletHandler.queueNextTaskOrFinish$javalin(JavalinServletHandler.kt:85) ~[javalin-4.6.1.jar:4.6.1]                                                                        
        at io.javalin.http.JavalinServlet.service(JavalinServlet.kt:89) ~[javalin-4.6.1.jar:4.6.1]                                                                                                            
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[jakarta.servlet-api-4.0.3.jar:4.0.3]                                                                                                
        at io.javalin.jetty.JavalinJettyServlet.service(JavalinJettyServlet.kt:58) ~[javalin-4.6.1.jar:4.6.1]                                                                                                 
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[jakarta.servlet-api-4.0.3.jar:4.0.3]                                                                                                
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) ~[jetty-servlet-9.4.46.v20220331.jar:9.4.46.v20220331]                                                                      
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:550) ~[jetty-servlet-9.4.46.v20220331.jar:9.4.46.v20220331]                                                                  
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]                                                            
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]                                                           
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]                                                            
        at io.javalin.jetty.JettyServer$start$wsAndHttpHandler$1.doHandle(JettyServer.kt:52) ~[javalin-4.6.1.jar:4.6.1]                                                                                       
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]                                                             
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501) ~[jetty-servlet-9.4.46.v20220331.jar:9.4.46.v20220331]                                                                   
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:179) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487) ~[jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732) [jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479) [jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) [jetty-server-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [jetty-io-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [jetty-io-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [jetty-io-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034) [jetty-util-9.4.46.v20220331.jar:9.4.46.v20220331]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_291]
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rfyu**
## Why are the changes needed?

Close #2614.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java [hunk 1]
```diff
@@ -364,13 +368,24 @@ public List<PartitionFileBaseInfo> getSnapshotDetail(AmoroTable<?> amoroTable, l
...
-                        arcticTable.spec().partitionToPath(f.partition()),
+                        getPartitionSpecBySpecId(arcticTable, f.specId())
+                            .partitionToPath(f.partition()),
                         f.path().toString(),
                         f.fileSizeInBytes(),
                         "remove")));
     return result;
   }
 
+  private PartitionSpec getPartitionSpecBySpecId(ArcticTable arcticTable, Integer fileSpecId) {
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java:379]: You can reuse `com.netease.arctic.utils.ArcticTableUtil#getArcticTablePartitionSpecById`.

**Review by zhoujinsong — COMMENTED** [review]
@rfyu Thanks a lot for your contribution!

I left some comments, PTAL.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java (hunk 1 above)
Comment (majin1102) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/MixedAndIcebergTableDescriptor.java:379]: > You can reuse `com.netease.arctic.utils.ArcticTableUtil#getArcticTablePartitionSpecById`.

Agree

**Review by majin1102 — APPROVED** [review]
LGTM
